src/core/storage/storage.js

```diff
@@ -152,6 +156,20 @@ export default class Storage {
     this.persistentStorage.delete(key);
   }
 
+  /**
+   * Returns the url representing the current persisted state, merged
+   * with any additional query params currently in the url.
+   *
+   * @returns {string}
+   */
+  getCurrentStateUrlMerged () {
+    const searchParams = new SearchParams(window.location.search.substring(1));
```

Comment:
I'm worried about having this in the function instead of passed into the function. What if we get into the situation where someone is changing the URL in between calls and we get two different Navigation tab URLs? COuld we move it to a parameter? I feel like it would make the test easier too

Reply:
As we talked about online, this is the behavior of the SDK currently. This is something we should address but not necessarily in this PR as it is not a breaking change.